### ISSUE TITLE
Unable to change email, OTP verification fails. "User not found"

### ISSUE THREAD

**Issue by loyoliteabid** [supabase/supabase#18223]
# Bug report
 When a signed user attempts to change their email address. OTP receives to the new email. however when we try to verifyOtp I get an error saying "User not found". This is happening only when we try to change email using type: email_change. other types like signup are working fine.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Signup using email 
2. Login using email and password
3. change email using 
   `const { data, error } = await supabase.auth.updateUser({ email });`
4. Verify otp using 
   `const { data, error } = await supabase.auth.verifyOtp({
      token: otp,
      type: "email_change",
      email, /* new email */
    });`
5. See error
 `{"name":"AuthApiError","message":"User not found","status":404}`

## Expected behavior
Signed user should be able to change their email with OTP verification

## System information

-Browser  chrome
- "@supabase/auth-helpers-nextjs": "latest",
- "@supabase/supabase-js": "latest",

**Comment by binaryArrow** [comment]
did you try it with `type: 'email'` instead of 'email_change'. For what exactly is the type 'email_change' used for?
I cannot find any explanation about this type in the docs.

**Comment by abiddraws** [comment]
email_change is mentioned here https://supabase.com/docs/reference/javascript/auth-verifyotp , you can reproduce the issue from this code 

`"use client";
import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
import { useState } from "react";

export default function EmailChange() {
  const supabase = createClientComponentClient();

  const [email, setEmail] = useState("");
  const [otp, setOtp] = useState("");
  const [otpSent, setOtpSent] = useState(false);

  const handleSendOTP = async (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    e.stopPropagation();
    const { data, error } = await supabase.auth.updateUser({ email });

    if (error) {
      console.log("Error from updateEmail ", JSON.stringify(error));
    } else {
      console.log("Email updation success ", JSON.stringify(data));
      setOtpSent(true);
    }
  };

  const handleVerifyOTP = async (e: React.MouseEvent<HTMLButtonElement>) => {
    console.log("Call receives...");
    e.preventDefault();
    e.stopPropagation();

    const { data, error } = await supabase.auth.verifyOtp({
      token: otp,
      type: "email_change",
      email,
    });

    if (error) {
      console.log(JSON.stringify(error));
    }
    if (data?.user) {
      alert(`OTP ${otp} verified successfully!`);
    }
  };

  return (
    <div className="max-w-md mx-auto p-6 border rounded-lg shadow-lg mt-10 bg-gray-50">
      <h2 className="text-2xl font-bold mb-4">OTP Verification</h2>

      <div className="mb-4">
        <label className="block text-gray-600">Email Address:</label>
        <input
          type="email"
          placeholder="Enter your email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          className="w-full p-2 border rounded-md focus:outline-none focus:ring focus:border-blue-300"
        />
      </div>

      {!otpSent ? (
        <button
          onClick={handleSendOTP}
          className="w-full bg-blue-500 text-white py-2 rounded-md hover:bg-blue-600 focus:outline-none focus:ring focus:border-blue-300"
        >
          Send OTP
        </button>
      ) : (
        <div>
          <div className="mb-4">
            <label className="block text-gray-600">Enter OTP:</label>
            <input
              type="text"
              placeholder="Enter OTP"
              value={otp}
              onChange={(e) => setOtp(e.target.value)}
              className="w-full p-2 border rounded-md focus:outline-none focus:ring focus:border-blue-300"
            />
          </div>
          <button
            onClick={handleVerifyOTP}
            className="w-full bg-blue-500 text-white py-2 rounded-md hover:bg-blue-600 focus:outline-none focus:ring focus:border-blue-300"
          >
            Verify OTP
          </button>
        </div>
      )}
    </div>
  );
}`

**Comment by binaryArrow** [comment]
from looking at the docs, it does not exactly explain what it is used for, but by looking at your code, you might maybe refactor this snippet:  `const { data, error } = await supabase.auth.verifyOtp({
      token: otp,
      type: "email_change",
      email,
    });
`
to this: 
`const { data, error } = await supabase.auth.verifyOtp({
      token: otp,
      type: "email_change",
      email: email,
    });
`
because you do not use the properties in the order they are documented, I guess you have to specify the property names...

**Comment by abiddraws** [comment]
I have tried 

```
const { data, error } = await supabase.auth.verifyOtp({
      token: otp,
      type: "email_change",
      email: email,
    });
```
    
    and also tried this 
    
   ```
 const { data, error } = await supabase.auth.verifyOtp({
      token: otp,
      type: 'email',
      email: email,
    });
```
    
    Still same error

**Comment by binaryArrow** [comment]
Did you run the code with a User you know is definitely in the Users Table in Supabase? -> Does it still fails then?

**Comment by abiddraws** [comment]
Yes the user is in the supabase auth/user table. and it fails while `verifyOtp`
